cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -436,6 +436,13 @@ Status ORCFileReader::Open(const std::shared_ptr<io::RandomAccessFile>& file,
   return Status::OK();
 }
 
+Result<std::unique_ptr<ORCFileReader>> ORCFileReader::Open(
+    const std::shared_ptr<io::RandomAccessFile>& file, MemoryPool* pool) {
+  auto result = std::unique_ptr<ORCFileReader>(new ORCFileReader());
+  RETURN_NOT_OK(result->impl_->Open(file, pool));
+  return std::move(result);
+}
+
```

Comment:
Now, the deprecated overloads can be rewritten to call the non-deprecated ones:
```c++
Status ORCFileReader::Open(const std::shared_ptr<io::RandomAccessFile>& file,
                           MemoryPool* pool, std::unique_ptr<ORCFileReader>* reader) {
  return Open(file, pool).Value(&reader);
}
```

Reply:
Wow, this is brilliant!
I guess it should be `Open(file, pool).Value(reader);`, right?

Reply:
Oh, you're right indeed.